lapis2-docs/src/components/ConfigGenerator/MetadataWizard.tsx

```diff
@@ -0,0 +1,219 @@
+            <div className='modal-box'>
+                <h3 className='font-bold text-lg'>Edit metadata</h3>
+                <div className='flex flex-col space-x-2 items-center'>
+                    <table className='table'>
+                        <tbody>
+                            <tr>
+                                <td>Name</td>
+                                <td>
+                                    <EditMetadataName metadataName={metadataName} setMetadataName={setMetadataName} />
+                                </td>
+                            </tr>
+
+                            <tr>
+                                <td>Type</td>
+                                <td>
+                                    <SelectMetadataType metadataType={metadataType} setMetadataType={setMetadataType} />
+                                </td>
+                            </tr>
+
+                            <tr>
+                                <td>Generate Index</td>
+                                <td>
+                                    <EditGenerateIndex
+                                        generateIndex={generateIndex}
+                                        setGenerateIndex={setGenerateIndex}
+                                    />
+                                </td>
+                            </tr>
+                        </tbody>
+                    </table>
+                </div>
+                <form method='dialog'>
+                    <div className='flex justify-end'>
+                        <button
+                            className='btn join-item'
+                            onClick={() => {
+                                handleUpdateMetadata();
+                            }}
+                        >
+                            OK
+                        </button>
+                        <button className='btn btn-ghost join-item'>Cancel</button>
+                    </div>
+                </form>
+            </div>
+        </dialog>
+    );
+}
+
+function EditGenerateIndex({
+    generateIndex,
+    setGenerateIndex,
+}: {
+    generateIndex: boolean;
+    setGenerateIndex: Dispatch<SetStateAction<boolean>>;
+}) {
+    const handleGenerateIndexChange = () => {
+        setGenerateIndex(!generateIndex);
+    };
+
+    return (
+        <input
+            type='checkbox'
+            className='toggle toggle-accent'
+            checked={generateIndex}
+            onChange={handleGenerateIndexChange}
+        />
+    );
+}
+
+function EditMetadataName({
+    metadataName,
+    setMetadataName,
+}: {
+    metadataName: string;
+    setMetadataName: Dispatch<SetStateAction<string>>;
+}) {
+    const { config } = useContext(ConfigContext);
+    const [error, setError] = useState(false);
+
+    const handleMetadataNameChange = (event: { target: { value: any } }) => {
+        if (config.metadata.filter((metadata) => metadata.name === event.target.value).length > 0) {
+            setError(true);
```

Comment:
I think this should not show an error if the name is taken by the currently edited field.